source/portable/mbedtls/core_pkcs11_mbedtls.c

```diff
@@ -3930,15 +3926,7 @@ CK_DECLARE_FUNCTION( CK_RV, C_SignInit )( CK_SESSION_HANDLE hSession,
                 case CKM_RSA_PKCS:
                 case CKM_ECDSA:
 
-                    /* See explanation in prvCheckValidSessionAndModule for this exception. */
-                    /* coverity[misra_c_2012_rule_10_5_violation] */
-                    if( xIsPrivate != ( CK_BBOOL ) CK_TRUE )
-                    {
-                        LogError( ( "Failed to initialize sign operation. Sign "
-                                    "operation attempted with private key." ) );
-                        xResult = CKR_KEY_TYPE_INCONSISTENT;
-                    }
-                    else if( ( pxSession->xSignKeyHandle == CK_INVALID_HANDLE ) || ( pxSession->xSignKeyHandle != hKey ) )
+                    if( ( pxSession->xSignKeyHandle == CK_INVALID_HANDLE ) || ( pxSession->xSignKeyHandle != hKey ) )
```

Comment:
Why is it okay to delete lines 3935-3940 here?  `xIsPrivate`  may have been changed to `false` by `PKCS11_PAL_GetObjectValue()` on line 3890.

Reply:
The same check is already there in 3907, it was getting checked twice. If `xIsPrivate` is false the if statement won't be entered.